### PR TITLE
8172095: Let Node.managed become CSS-styleable

### PR DESCRIPTION
**Pull request by abhinayagarwal**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8172095](https://bugs.openjdk.java.net/browse/JDK-8172095): Let Node.managed become CSS-styleable

### Reviewers
 * @Dlittler949 (no known github.com user name / role) 🔄 Re-review required (review applies to 56b8f24b94386d5d808f5736e3bc1b6d2ecb2353)
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Ambarish Rapte](https://openjdk.java.net/census#arapte) (@arapte - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/602/head:pull/602` \
`$ git checkout pull/602`

Update a local copy of the PR: \
`$ git checkout pull/602` \
`$ git pull https://git.openjdk.java.net/jfx pull/602/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 602`

View PR using the GUI difftool: \
`$ git pr show -t 602`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/602.diff">https://git.openjdk.java.net/jfx/pull/602.diff</a>

</details>

### CODE REVIEW

**Review by Dlittler949 — APPROVED** [review]

**Comment by kevinrushforth** [comment]
This will need at least a brief discussion on the openjfx-dev mailing list.

/csr
/reviewers 2

**Comment by kevinrushforth** [comment]
Please start a new thread and reference that one. That thread asks for a new property, but I think what you propose, which is to make the existing `managed` property CSS styleable, is better. Given how simple a request yours is, I expect the discussion will be brief.

**Comment by abhinayagarwal** [comment]
Thanks for the feedback. I will create a CSR and start a thread on the mailing list.

**Comment by abhinayagarwal** [comment]
I started a [discussion on the mailing list](https://mail.openjdk.java.net/pipermail/openjfx-dev/2021-August/031520.html). Is there anything else pending to move this PR forward?

**Comment by kevinrushforth** [comment]
> Is there anything else pending to move this PR forward?

Nothing else needed for now. The review can proceed.

File: modules/javafx.graphics/src/main/docs/javafx/scene/doc-files/cssref.html [hunk 1]
```diff
@@ -1750,6 +1750,13 @@ <h4><a id="node">Node</a></h4>
           <td>See <a href="http://www.w3.org/TR/CSS2/visufx.html#visibility">W3C
               visibility property</a></td>
         </tr>
+        <tr>
+            <th class="propertyname" scope="row">-fx-managed</th>
+            <td class="value"><a href="#typeboolean" class="typelink"><boolean></a></td>
```
Comment (kevinrushforth) [inline modules/javafx.graphics/src/main/docs/javafx/scene/doc-files/cssref.html:1755]: This isn't rendered correctly. You need to use `&lt;boolean&gt;`.

**Review by kevinrushforth — COMMENTED** [review]
The new API looks fine. I left one comment on the cssref docs, and that's otherwise ready to go (so you can get the CSR ready).

I noticed that there are no existing tests in `NodeTest` for the default value or for setting / getting the property, which there are for other properties. Can you add these missing tests, to verify that there are no problems?

**Review by arapte — APPROVED** [review]
Changes look good to me. 
The tests, managedSetFromCSS() and Node_cssMethods_Test fail before and pass after. Other tests do pass both before and after, which is expected behaviour.

**Review by kevinrushforth — APPROVED** [review]
Looks good.

**Comment by kevinrushforth** [comment]
This is ready to be integrated once the CSR is approved.

**Pull request closed without merging**